File: CCC/RankingSystem/level3/src/submission.rs
use crate::time::Time;

#[derive(Debug, Clone, Eq, PartialEq)]
pub struct Submission {
    pub id: u32,
    pub time: Time,
    pub correct: bool,
}

impl Submission {
    pub fn new(id: u32, time: Time, correct: bool) -> Submission {
        Submission { id, time, correct }
    }
}

impl PartialOrd for Submission {
    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
        Some(self.cmp(other))
    }
}

impl Ord for Submission {
    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
        self.time.cmp(&other.time)
    }
}

File: CCC/RankingSystem/RankingSystem/Submission.cs
public class Submission : IComparable<Submission>
{
    public int Id { get; set; }
    public Time Time { get; set; }
    public bool Correct { get; set; }
    public int TaskId { get; set; }

    public Submission(int id, string time, bool correct, int taskId)
    {
        Id = id;
        Time = new Time(time);
        Correct = correct;
        TaskId = taskId;
    }

    public override string ToString()
    {
        return $"Id: {Id}, Time: {Time}, Correct: {Correct}, TaskId: {TaskId}";
    }

    public int CompareTo(Submission? other)
    {
        return Time.CompareTo(other?.Time);
    }
}